include/model.php

```diff
@@ -702,34 +702,25 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
 	 * @phpstan-return list<PLL_Language>
 	 */
 	protected function get_languages_from_taxonomies() {
-		$terms_by_type = $this->get_language_terms_by_type();
 
-		if ( empty( $terms_by_type['post'] ) || empty( $terms_by_type['term'] ) ) {
-			// `post` and `term` languages are mandatory.
-			return array();
-		}
+		// To get language taxonomies first.
+		$reversed_terms = array_reverse( $this->get_language_terms() );
```

Comment:
Is it really necessary? Since terms are ordered with keys, not sure we have to do it.

Reply:
Also, they are ordered by `term_group` which already order language right?

Reply:
In most case, it's right because all `term_group` are set to `0`.
However it exists one case where it's wrong: when you order the languages with the `term_group` (`order` column  in the languages list UI). `term_language` taxonomy terms have always `term_group` set to `0`

Let's take an example with 2 languages (en and fr):

`language` taxonomy term `en` with `term_group` set to `2`
`language` taxonomy term `fr` with `term_group`  set to `1`
`term_language taxonomy term `pll_en` with `term_group` set to `0`
`term_language taxonomy term `pll_fr` with `term_group` set to `0`

So the terms request ordered by `term_group` returns:

pll_en (term_group = 0)
pll_fr (term_group = 0)
fr (term_group = 1)
en (term_group = 2)

At the end we can't ensure the languages order because we start the process with the English before the French as we expect.

`array_reverse` is a concise manner to put the language taxonomy terms at the beginning of the results.
As you put these terms in the reverse order compared to the `term_group` order, we need to also reverse the output array. Thus the language taxonmy terms are in the right order.

Reply:
OK I see now, thanks for the explanation.